meson.build

```diff
@@ -57,88 +70,93 @@ else
-  if opencv.found()
-      add_project_arguments('-DUSE_OPENCV', language : 'cpp')
-  else
-    opencv = dependency('opencv4', method : 'pkg-config', required: false)
-    if opencv.found()
-      add_project_arguments('-DUSE_OPENCV', language : 'cpp')
-    else
+  if not opencv.found()
+    opencv = dependency('opencv4', required: false)
+    if not opencv.found()
       error('OpenCV not found but webcam build requested')
     endif
   endif
-else
-  opencv=dependency('',required: false)
 endif
 
-#Portmidi is maybe not found in pkg-config, use find library as an alternative
+# Portmidi is maybe not found in pkg-config, use find library as an alternative,
+# with both names libportmidi (on ubuntu) and portmidi (on OSX)
+portmidi = nulldependency
 if get_option('usemididrum')
-  message('Portmidi build requested')
   portmidi = dependency('portmidi', required: false)
-  if portmidi.found()
-    add_project_arguments('-DUSE_PORTMIDI', language: 'cpp')
-  else
+  if not portmidi.found()
     portmidi = compiler.find_library('libportmidi', required: false)
-    if portmidi.found()
-      add_project_arguments('-DUSE_PORTMIDI', language: 'cpp')
-    else
-      error('PORTMIDI not found but MIDI drum support build requested')
+    if not portmidi.found()
+      portmidi = compiler.find_library('portmidi', required: false)
+      if not portmidi.found()
+        error('PORTMIDI not found but MIDI drum support build requested')
+      endif
     endif
   endif
-else
-  portmidi=dependency('',required: false)
 endif
 
-#Cpprest may not be in pkg-config
-#Also, libssl and libcrypto are only needed when webserver support is requested
+# Note: cpprest may not be in pkg-config
+cpprestneededandfound = false
+cpprest = nulldependency
 if get_option('usewebserver')
-  message('Webserver build requested')
-  cpprest=dependency('cpprest', required : false)
+  cpprest = dependency('cpprest', required: false)
   if cpprest.found()
-    add_project_arguments('-DUSE_WEBSERVER', language: 'cpp')
+    cpprestneededandfound = true
   else
-    cpprest=compiler.find_library('libcpprest', required:false)
+    cpprest = compiler.find_library('libcpprest', required: false)
     if cpprest.found()
-      add_project_arguments('-DUSE_WEBSERVER', language: 'cpp')
+      cpprestneededandfound = true
     else
-      error('CppRestSdk not found but webserver build requested')
+      cpprest = compiler.find_library('cpprest', required: false)
+      if cpprest.found()
+        cpprestneededandfound = true
+      else
+        error('CppRestSdk not found but webserver build requested')
+      endif
     endif
   endif
-else
-  cpprest=dependency('',required: false)
 endif
 
-#Write configuration file
-configure_file(input : 'game/config.hh.in', output : 'config.hh', configuration : conf)
+libssl = nulldependency
+libcrypto = nulldependency
+
+if cpprestneededandfound
+  libssl = dependency('libssl', required: true)
+  libcrypto = dependency('libcrypto', required: true)
```

Comment:
I think this can be moved into the `if get_options('usewebserver')` as it's only needed within that

Reply:
Indeed, I think it could further simplify the script, so let me have a look at at.